### PR TITLE
fix: display custom schematics in help

### PR DESCRIPTION
**Pull request by daniel-samson**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/nestjs/nest/blob/master/CONTRIBUTING.md
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[x] Bugfix
[ ] Feature
[x] Code style update (formatting, local variables)
[x] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Other... Please describe:
```

## What is the current behavior?

Related:
Issue Number: #1433 
PR: #1434 

Currently, when you run `nest generate --help`, it will only display schematics from the `@nestjs/schematics` package. If you have a custom collection configured in `nest-cl.json`, it doesn't display the schematics from your custom package.

Example:

Screenshot 2023-02-02 at 22 05 56

## What is the new behavior?
Example:

Screenshot 2023-02-02 at 20 35 20

After this change, you can see which collection is being selected and a table of schematics from that collection

## Does this PR introduce a breaking change?
```
[ ] Yes
[X] No **
```

**It changes the commander to load async, and the getSchematics is no longer static, so sort of...

## Other information

I have manually tested the local command and the global command, and it seems to work. This change is important to me, as it allows me to easily train my team to use the custom schematics i have made.

 I am on the discord (verbing#6551)

### CODE REVIEW

File: lib/schematics/custom.collection.ts [hunk 1]
```diff
@@ -1,3 +1,30 @@
...
+  description: string;
+  aliases: string[];
+}
+
+export class CustomCollection extends AbstractCollection {
+  public getSchematics(): Schematic[] {
+    this.collection;
+    const collectionPackagePath = dirname(require.resolve(this.collection));
+    const collectionPath = join(collectionPackagePath, 'collection.json');
+    const collection = JSON.parse(readFileSync(collectionPath).toString());
```
Comment (micalevisk) [inline lib/schematics/custom.collection.ts:18]: what do we need to call `JSON.parse` here? is it for JSON validation?

File: lib/schematics/custom.collection.ts [hunk 2]
```diff
@@ -1,3 +1,30 @@
...
-export class CustomCollection extends AbstractCollection {}
+export interface CollectionSchematic {
+  schema: string;
+  description: string;
+  aliases: string[];
+}
+
+export class CustomCollection extends AbstractCollection {
+  public getSchematics(): Schematic[] {
+    this.collection;
```
Comment (micalevisk) [inline lib/schematics/custom.collection.ts:15]:
```suggestion
```

File: lib/schematics/custom.collection.ts (hunk 1 above)
Comment (micalevisk) [inline lib/schematics/custom.collection.ts:18]:
```suggestion
    const collection = JSON.parse(readFileSync(collectionPath, 'utf8'));
```

**Comment by daniel-samson** [comment]
@micalevisk Thank you for the comments. They were very helpful. I have made the change you have asked for. Let me know if you need me to make any more changes.

**Review by micalevisk — APPROVED** [review]

**Comment by kamilmysliwiec** [comment]
LGTM

**Pull request merged**